src/cache-utils.ts

```diff
@@ -95,3 +96,28 @@ export const getCacheDirectoryPath = async (
 
   return stdOut;
 };
+
+export function isGhes(): boolean {
+  const ghUrl = new URL(
+    process.env['GITHUB_SERVER_URL'] || 'https://github.com'
+  );
+  return ghUrl.hostname.toUpperCase() !== 'GITHUB.COM';
+}
+
+export function isCacheFeatureAvailable(): boolean {
+  if (!cache.isFeatureAvailable()) {
+    if (isGhes()) {
+      throw new Error(
+        'Cache action is only supported on GHES version >= 3.5. If you are on version >=3.5 Please check with GHES admin if Actions cache service is enabled or not.'
+      );
+    } else {
+      core.warning(
+        'An internal error has occurred in cache backend. Please check https://www.githubstatus.com/ for any ongoing issue in actions.'
```

Comment:
@dmitry-shibanov in https://github.com/actions/setup-python/pull/363 we ended up changing this warning:

```suggestion
        'The runner was not able to contact the cache service. Caching will be skipped'
```

Reply:
Done. Thank you.